src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -37,22 +38,33 @@ internal sealed class AggregatorStore
         private readonly AggregationTemporality temporality;
         private readonly bool outputDelta;
         private readonly MetricPoint[] metricPoints;
+        private readonly ReaderWriterLockSlim[] metricPointLocks;
+        private readonly int[] currentMetricPointBatch;
         private readonly AggregationType aggType;
         private readonly double[] histogramBounds;
         private readonly UpdateLongDelegate updateLongCallback;
         private readonly UpdateDoubleDelegate updateDoubleCallback;
-        private int metricPointIndex = 0;
+        private long batchSize = 0;
         private bool zeroTagMetricPointInitialized;
         private DateTimeOffset startTimeExclusive;
         private DateTimeOffset endTimeInclusive;
 
+        private ConcurrentStack<int> metricPointFreeList = new ConcurrentStack<int>(Enumerable.Range(1, MaxMetricPoints - 1));
```

Comment:
we could populate it in reverse order. so the index 1 will be the first to be used, then index 2.

Reply:
I've removed the free list.